libs/application/template-api-modules/src/lib/modules/templates/driving-license-submission/driving-license-submission.service.ts

```diff
@@ -28,10 +28,17 @@ export class DrivingLicenseSubmissionService {
   ) {}
 
   async createCharge({
-    application: { id },
+    application: { id, externalData },
     authorization,
   }: TemplateApiModuleActionProps) {
-    return this.sharedTemplateAPIService.createCharge(authorization, id)
+    const parsedPaymentData = JSON.parse(
+      JSON.stringify(externalData.payment.data),
+    )
```

Comment:
What is going on here:

```typescript
JSON.parse(JSON.stringify(...))
```

Reply:
I was getting undefined when trying to send the item specifically through externalData.payment.data.chargeItemCode.
Is there a better way to access such data point without risking undefined?

Reply:
f.x
`console.log(externalData.payment.data?.chargeItemCode)`
gives error :

> Property 'chargeItemCode' does not exist on type 'string | number | boolean | object'.
  Property 'chargeItemCode' does not exist on type 'string'.ts(2339)

Reply:
Yeah so this is because `externalData.data` is of type `string | number | boolean | object`.

I believe you'll just need to tell the typescript compiler what to expect here, do a type cast into the type you expect this to be (since this can be of virtually any type).

Something like:

```typescript
interface PaymentData {
  chargeItemCode: ....
}

// ...

const paymentData = externalData.payment.data as PaymentData
```